pkg/operation/botanist/controlplane.go

```diff
@@ -917,6 +909,19 @@ func (b *Botanist) DeployKubeAPIServer() error {
 				return err
 			}
 		}
+	} else {
+		// If HVPA is disabled, delete any HVPA that was already deployed
+		u := &unstructured.Unstructured{}
+		u.SetName(v1alpha1constants.DeploymentNameKubeAPIServer)
+		u.SetNamespace(b.Shoot.SeedNamespace)
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "autoscaling.k8s.io",
+			Version: "v1alpha1",
+			Kind:    "Hvpa",
+		})
+		if err := b.K8sSeedClient.Client().Delete(context.TODO(), u); client.IgnoreNotFound(err) != nil {
```

Comment:
I am running with

```yaml
featureGates:
  HVPA: false
  HVPAForShootedSeed: false
```

And I get 
```
lastOperation:
    description: 'Flow "Shoot cluster reconciliation" encountered task errors: [task
      "Deploying main and events etcd" failed: retry failed with context deadline
      exceeded, last error: no matches for kind "Hvpa" in version "autoscaling.k8s.io/v1alpha1"]
      Operation will be retried.'
```

Can we make possible running gcm when these features gate are off?

/cc @ggaurav10

Reply:
thanks @ialidzhikov for bringing this to notice. Fixed in #1678